### PR TITLE
Support Sarif output

### LINKED ISSUE #2866: idea: add sarif output format
### Your feature request related to a problem? Please describe.

[SARIF](http://docs.oasis-open.org/sarif/sarif/v2.0/csprd01/sarif-v2.0-csprd01.html) is becoming the de-facto industry standard for SAST scanning results. Several tools already support it, like for example gitlab, github and more.

When adding the golangci-lint action on github I was missing an option to choose SARIF as output format to allow github to import any findings into the security tab or as a check result set in pullrequests.

Using the action in its current form does not seem to satisfyingly link findings to their actual locations in the code.

### Describe the solution you'd like.

I'd like to see a SARIF output format (`output.format: sarif`) and ideally an option to define an output file name. 

### Describe alternatives you've considered.

* Using the action in its current form and live with the shortcomings. 
* Search for an alternative tool.

### Additional context.

_No response_

### PR DESCRIPTION
**Pull request by SophisticaSean**
This work supports exporting Sarif file format. It's a common enough format and I figured I'd help out by adding it. :)

Sarif documentation:
https://github.com/microsoft/sarif-tutorials/blob/main/docs/2-Basics.md#results
https://docs.oasis-open.org/sarif/sarif/v2.1.0/cs01/schemas/sarif-external-property-file-schema-2.1.0.json

fixes #2866

### CODE REVIEW

File: pkg/printers/sarif.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+const (
+	sarifVersion   = "2.1.0"
+	sarifSchemaURI = "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.4.json"
```
Comment (alexandear) [inline pkg/printers/sarif.go:12]: Maybe we should use the newest available version?
```suggestion
	sarifSchemaURI = "https://schemastore.azurewebsites.net/schemas/json/sarif-2.1.0-rtm.6.json"
```

File: pkg/printers/sarif.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+						},
+					},
+				},
+			},
+		}
+
+		run.Results = append(run.Results, sr)
+	}
+
+	output.Runs = []sarifRun{run}
```
Comment (alexandear) [inline pkg/printers/sarif.go:107]: I propose to move `output` below:
```suggestion
	run := sarifRun{}
	run.Tool.Driver.Name = "golangci-lint"

	for i := range issues {
		issue := issues[i]

		severity := issue.Severity
		if severity == "" {
			severity = "error"
		}

		sr := sarifResult{
			RuleID:  issue.FromLinter,
			Level:   severity,
			Message: sarifMessage{Text: issue.Text},
			Locations: []sarifLocation{
				{
					PhysicalLocation: sarifPhysicalLocation{
						ArtifactLocation: sarifArtifactLocation{URI: issue.FilePath()},
						Region: sarifRegion{
							StartLine:   issue.Line(),
							StartColumn: issue.Column(),
						},
					},
				},
			},
		}

		run.Results = append(run.Results, sr)
	}

	output := SarifOutput{
		Version: sarifVersion,
		Schema:  sarifSchemaURI,
		Runs:    []sarifRun{run},
	}
}
```

**Comment by SophisticaSean** [comment]
Wow thanks so much for the quick fixes and reviews. Y'all are impressive!

**Review by ldez — APPROVED** [review]
LGTM

**Pull request merged**